src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java

```diff
@@ -211,6 +302,34 @@ private static Field findField(Class<?> clazz, String name)
         }
     }
 
+    /**
+     * Finds a field with the specified index in the given class and makes it accessible.
+     * Note: For performance, store the returned value and avoid calling this repeatedly.
+     * <p>
+     * Throws an exception if the field is not found.
+     *
+     * @param clazz      The class to find the field on.
+     * @param fieldIndex The index of the field on the class
+     * @param <T>        The type.
+     * @return The constructor with the specified parameters in the given class.
+     * @throws NullPointerException       If {@code clazz} is null.
+     * @throws UnableToFindFieldException If the field could not be found.
+     * @deprecated Use {@link #findField(Class, String)} because field indices change a lot more often than field names do.
+     */
+    @Deprecated
+    @Nonnull
```

Comment:
Don't ADD a deprecated entry...

Reply:
I added it for completeness, but it shouldn't be used. Should all the methods that use indices be removed?

Reply:
They can't be removed, as that breaks binary compatibility. But it needs a @Deprecated //Remove in 1.15 marker.